risc0/core/src/field/baby_bear.rs

```diff
@@ -681,14 +695,11 @@ impl ops::MulAssign for ExtElem {
         //             = c_i * sum_j=0^(EXT_SIZE - 1) -IRREDUCIBLE_WITHOUT_LEADING_TERM * x^(i - EXT_SIZE + j)
         // So, we can zero out the x^i term by adding c[i] * -IRREDUCIBLE[j] to c[i - EXT_SIZE + j] for j < EXT_SIZE
         assert_eq!(IRREDUCIBLE.len(), EXT_SIZE + 1);
-        assert_eq!(IRREDUCIBLE[EXT_SIZE], Elem::new(1));
-        let upper = 2 * EXT_SIZE - 2;
-        for i_rev in (0..EXT_SIZE - 1).rev() {
-            // We need to iterate from high degree to low, otherwise we might add to coefficients we've already zeroed out
-            let i = upper - i_rev;
-            for j in 0..(IRREDUCIBLE.len() - 1) {
-                // Perhaps there's a minor perf gain to be had by storing NEG_IRREDUCIBLE instead of IRREDUCIBLE itself
-                c[i - EXT_SIZE + j] += c[i] * -IRREDUCIBLE[j];
+        assert_eq!(IRREDUCIBLE[EXT_SIZE], Elem::ONE);
+        // We need to iterate from high degree to low, otherwise we might add to coefficients we've already zeroed out.
+        for i in EXT_SIZE..2 * EXT_SIZE - 1 {
```

Comment:
This comment and this loop index contradict each other. For our choice of irreducible polynomial (and in general any irreducible polynomial of the form `x^EXT_SIZE + constant`) it's ok to iterate from low degree to high as coded here, but we should update the comment to match the code.
```suggestion
        // Since our irred. poly is `x^EXT_SIZE + c` it is ok to iterate from low degree (small `i`) to high
        for i in EXT_SIZE..2 * EXT_SIZE - 1 {
```